app/src/main/java/com/keylesspalace/tusky/components/compose/ComposeViewModel.kt

```diff
@@ -149,29 +152,34 @@ class ComposeViewModel @Inject constructor(
                             item.copy(id = event.mediaId, uploadPercent = -1)
                     }
                     synchronized(media) {
-                        val mediaValue = media.value!!
-                        val index = mediaValue.indexOfFirst { it.localId == newMediaItem.localId }
-                        media.postValue(
-                            if (index == -1) {
-                                mediaValue + newMediaItem
-                            } else {
-                                mediaValue.toMutableList().also { it[index] = newMediaItem }
-                            }
-                        )
+                        val index = media.value.indexOfFirst { it.localId == newMediaItem.localId }
+                        media.value = if (index == -1) {
+                            media.value + newMediaItem
+                        } else {
+                            media.value.toMutableList().also { it[index] = newMediaItem }
+                        }
                     }
                 }
         }
-        return mediaItem
+        mediaItem
     }
 
     private fun addUploadedMedia(id: String, type: QueuedMedia.Type, uri: Uri, description: String?) {
-        val mediaItem = QueuedMedia(System.currentTimeMillis(), uri, type, 0, -1, id, description)
-        media.value = media.value!! + mediaItem
+        val mediaItem = QueuedMedia(
+            localId = (media.value.maxOfOrNull { it.localId } ?: 0) + 1,
+            uri = uri,
+            type = type,
+            mediaSize = 0,
+            uploadPercent = -1,
+            id = id,
+            description = description
+        )
+        media.value = media.value + mediaItem
```

Comment:
why do we `synchronized(media)` in one place but not all other places where we modify it? Isn't it defeating the point a bit? There's also [Mutex](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.sync/-mutex/) which has nifty [withLock](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.sync/with-lock.html) (it's kind of a lame lock that doesn't contain data but still better than nothing)